source/grid/grid_generator.cc

```diff
@@ -1454,74 +1454,75 @@ namespace GridGenerator
       if (tria.n_cells() != 3)
         AssertThrow(false, ExcNotImplemented());
 
-      double middle = (outer_radius - inner_radius) / 2e0 + inner_radius;
-      double eps    = 1e-3 * middle;
-      Triangulation<3>::cell_iterator cell = tria.begin();
+      const double middle = (outer_radius - inner_radius) / 2e0 + inner_radius;
+      const double eps    = 1e-3 * middle;
 
-      for (; cell != tria.end(); ++cell)
+      for (const auto &cell : tria.cell_iterators())
         for (const unsigned int f : GeometryInfo<3>::face_indices())
           {
-            if (!cell->face(f)->at_boundary())
+            const auto face = cell->face(f);
+            if (!face->at_boundary())
               continue;
 
-            double radius = cell->face(f)->center().norm() - center.norm();
-            if (std::fabs(cell->face(f)->center()[0]) <
-                eps) // x = 0 set boundary 2
+            // One could use the radius of the center of the face, but for
+            // curved cells the radius of the (euclidean) center is not
+            // equal to the radius at the vertices. Compute the average
+            // radius of all vertices instead.
+            double radius = 0.0;
+            for (unsigned int v : face->vertex_indices())
+              radius += (face->vertex(v) - center).norm();
+
+            radius /= face->n_vertices();
+
+            if (std::fabs(face->center()[0]) < eps) // x = 0 set boundary 2
               {
-                cell->face(f)->set_boundary_id(2);
+                face->set_boundary_id(2);
                 for (unsigned int j = 0; j < GeometryInfo<3>::lines_per_face;
                      ++j)
```

Comment:
```suggestion
                for (const unsigned int line_no : face->line_indices())
```
if we are using the range-for above then we can improve consistency by changing this too.